src/scanner/custom-rules/display-accessible-names.ts

```diff
@@ -14,55 +15,30 @@ export const accessibleNamesConfiguration: RuleConfiguration = {
         enabled: false,
         any: [accessibleNamesCheckId],
+        matches: hasAccessibleName,
     },
 };
 
-function createSelector(): string {
-    const roles = [
-        'alertdialog',
-        'application',
-        'button',
-        'checkbox',
-        'columnheader',
-        'combobox',
-        'dialog',
-        'grid',
-        'heading',
-        'img',
-        'link',
-        'listbox',
-        'marquee',
-        'meter',
-        'menuitem',
-        'menuitemcheckbox',
-        'menuitemradio',
-        'option',
-        'progressbar',
-        'radio',
-        'radiogroup',
-        'region',
-        'rowheader',
-        'searchbox',
-        'slider',
-        'spinbutton',
-        'switch',
-        'table',
-        'tabpanel',
-        'textbox',
-        'tooltip',
-        'tree',
-        'treegrid',
-        'treeitem',
-    ];
-
-    const selectors: string[] = [];
-    roles.forEach((role: string) => {
-        selectors.push('[role=' + role + ']', role);
-    });
-    return selectors.join(',');
+function hasAccessibleName(node: HTMLElement): boolean {
+    const nameProhibitedSelectors: string =
+        'caption, figcaption, [role=caption], code, [role=code], del, [role=deletion], em, [role=emphasis],[role=generic], ins, [role=insertion], p, [role=paragraph], [role=presentation], [role=strong], strong, sub, sup, [role=subscript], [role=superscript], [role=none]';
+    if (axe.utils.matchesSelector(node, nameProhibitedSelectors)) {
+        return false;
+    }
+    if (AxeUtils.getAccessibleText(node) === '') {
+        return false;
+    }
+    if (
+        (node.tagName.toLowerCase() === 'div' || node.tagName.toLowerCase() === 'span') &&
```

Comment:
Will this condition ever return true? Looking at the `selector` property above, `div` and `span` elements with no role do not seem to be included in this rule